packages/backend/src/backend_factory.ts

```diff
@@ -55,7 +56,15 @@ export class BackendFactory<
    * Initialize an Amplify backend with the given construct factories and in the given CDK App.
    * If no CDK App is specified a new one is created
    */
-  constructor(constructFactories: T, stack: Stack = createDefaultStack()) {
+  constructor(
+    constructFactories: T,
+    stack?: Stack,
+    mainStackProps?: MainStackProps
```

Comment:
`stack` and `mainStackProps` should be mutually exclusive.

Reply:
Have added constructor overloads to ensure this exclusivity in [3b24347](https://github.com/aws-amplify/amplify-backend/pull/1675/commits/3b24347294b12115cdb8ca80f822d89fd8706169)